src/providers/Data/Data.provider.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react';
+
+const DataContext = React.createContext({});
+
+function DataProvider({ children }) {
+  const [apiResponse, setApiResponse] = useState([]);
+
+  useEffect(() => {
+    const abortController = new AbortController();
+    const { signal } = abortController;
+    const API =
+      'https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=12&q=popular&type=video&key=AIzaSyAJo6nlfpXltr75BUwKhCabevhoDK4CndM';
```

Comment:
instead of having your API key here, you could extract that kind of values to environment values used at running/build time.  Check ".env.local" file or `dotenv` node library for this

Reply:
For reference: https://www.npmjs.com/package/dotenv